### PR TITLE
SDIT-776: 🐛 Fixes for business hours and test

### PR DESCRIPTION
**Pull request by petergphillips**

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -2,52 +2,30 @@
 
 default: build
 
-charts/generic-prometheus-alerts/ci/rules.yaml: charts/generic-prometheus-alerts/ci/compiled-yaml.yaml
-	@echo "Extracting Rules ..."
-	@cd charts/generic-prometheus-alerts/ci && yq eval 'select(.metadata.name == "test-application-rules") | .spec' compiled-yaml.yaml > rules.yaml
+ci_dir="charts/generic-prometheus-alerts/ci"
+objects=rules app ingress rds sns sqs
+apps=test-application test-business-hours
```
Comment (petergphillips) [inline Makefile:7]: we need two apps in order to test properly - one where alerts are only enabled during business hours and the other when they're enabled all of the time

File: charts/generic-prometheus-alerts/templates/application-alerts.yaml [hunk 2]
```diff
@@ -7,8 +7,8 @@
 {{- $targetJobName := default $targetApplicationRegex .Values.jobTargetOverride }}
 {{- $targetCronjob := default $targetApplicationRegex .Values.cronjobTargetOverride }}
 {{- $targetHpa := default .Values.targetApplication .Values.hpaTargetOverride }}
-{{- $targetApplicationUnderscore := .Values.targetApplication | replace "-" "_" }}
-{{- $businessOrAllHoursExpression := ternary "and ON() {{ $targetApplicationUnderscore }}:business_hours" "" .Values.businessHoursOnly}}
```
Comment (petergphillips) [inline charts/generic-prometheus-alerts/templates/application-alerts.yaml:11]: this wasn't expanded by helm so ended up being in prometheus query and therefore keeling over

File: charts/generic-prometheus-alerts/templates/rules.yaml [hunk 3]
```diff
@@ -1,3 +1,5 @@
+{{- if .Values.businessHoursOnly -}}
```
Comment (petergphillips) [inline charts/generic-prometheus-alerts/templates/rules.yaml:1]: no point creating rules if we don't need to

File: charts/generic-prometheus-alerts/templates/application-alerts.yaml (hunk 2 above)
Comment (mattops) [reply charts/generic-prometheus-alerts/templates/application-alerts.yaml:11]: I should have spotted that before-  the fix looks good.

**Review by mattops — APPROVED** [review]

**Pull request merged**